app/controllers/application_controller.rb

```diff
@@ -2,4 +2,29 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  # before_action :authenticate_user!
+  def custom_authentication
+    I18n.locale = params[:locale]
+
+    if (user_signed_in?)
```

Comment:
instead of this `if...else...end` block use `raise AccessDenied unless user_signed_in?`